packages/extension-ui/src/components/ValidatedInput.tsx

```diff
@@ -0,0 +1,57 @@
+// Copyright 2019 @polkadot/extension-ui authors & contributors
+// This software may be modified and distributed under the terms
+// of the Apache-2.0 license. See the LICENSE file for details.
+
+import React, { useState, useEffect } from 'react';
+
+import styled from 'styled-components';
+import { Result, Validator } from '@polkadot/extension-ui/validators';
+
+interface Props {
+  className?: string;
+  validator: Validator<string>;
+  children: React.ReactElement<{
+    isError: boolean;
+    value: string;
+    onChange: (value: string) => void;
+  }>;
+  onChange: (value: string | null) => void;
+}
+
+function ValidatedInput ({ className, validator, children, onChange }: Props): React.ReactElement<Props> {
+  const [value, setValue] = useState('');
+  const [wasMounted, setWasMounted] = useState(false);
+  const [validationResult, setValidationResult] = useState<Result<string>>(Result.ok(''));
+
+  useEffect(() => {
+    if (!wasMounted) {
+      setWasMounted(true);
+      return;
+    }
+    (async (): Promise<void> => {
+      const result = await validator(value);
+      setValidationResult(result);
+      onChange(Result.isOk(result) ? value : null);
+    })();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [value, validator, onChange]);
+
+  return (
+    <div className={className}>
+      {React.cloneElement(children, { isError: Result.isError(validationResult), value, onChange: setValue })}
+      {Result.isError(validationResult) && <ErrorMessage>{validationResult.error.errorDescription}</ErrorMessage>}
```

Comment:
Is there no other way of doing this? (The clone & isError React items)

Reply:
We could pass callback with error, value and callback. Actually tried doing it that way at first, but it was much uglier and less readable

Reply:
Really uncomfortable with this approach, especially around `clone` i.e. next stable release of React recommends `React.jsx(...)` as opposed to `React.createElement(...)`, so dunno what will happen going forward.

There could always be a hook that can create all these callbacks and then passed as-is (so 1 value instead of 3 of them - so the user has a single line to take care of and the magic happens here)

Reply:
Removed `cloneElement`